ambry-cloud/src/main/java/com/github/ambry/cloud/AzureStoreManager.java

```diff
@@ -0,0 +1,231 @@
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package com.github.ambry.cloud;
+
+import com.azure.core.http.rest.PagedIterable;
+import com.azure.storage.blob.BlobContainerClient;
+import com.azure.storage.blob.BlobServiceClient;
+import com.azure.storage.blob.models.BlobContainerItem;
+import com.azure.storage.blob.models.BlobErrorCode;
+import com.azure.storage.blob.models.BlobStorageException;
+import com.codahale.metrics.MetricRegistry;
+import com.github.ambry.cloud.azure.AzureCloudConfig;
+import com.github.ambry.cloud.azure.AzureMetrics;
+import com.github.ambry.cloud.azure.ConnectionStringBasedStorageClient;
+import com.github.ambry.clustermap.ClusterMap;
+import com.github.ambry.clustermap.PartitionId;
+import com.github.ambry.clustermap.ReplicaId;
+import com.github.ambry.config.CloudConfig;
+import com.github.ambry.config.VerifiableProperties;
+import com.github.ambry.server.ServerErrorCode;
+import com.github.ambry.server.StoreManager;
+import com.github.ambry.store.Store;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Storage manager that tracks {@link AzureStore} objects.
+ */
+public class AzureStoreManager implements StoreManager {
+  private static final Logger logger = LoggerFactory.getLogger(AzureStoreManager.class);
+  protected final ConcurrentHashMap<PartitionId, AzureStore> partitionToAzureStore;
+  protected final MetricRegistry metricRegistry;
+  protected final CloudConfig cloudConfig;
+  protected final AzureCloudConfig azureCloudConfig;
+  protected final AzureMetrics azureMetrics;
+  protected final BlobServiceClient azureStorageClient;
+  protected final VerifiableProperties properties;
+  protected final ClusterMap clusterMap;
+
+  /**
+   * Constructor
+   * @param properties Configuration parameters
+   * @param metricRegistry Metrics
+   * @param clusterMap Cluster-map
+   */
+  public AzureStoreManager(VerifiableProperties properties, MetricRegistry metricRegistry, ClusterMap clusterMap) {
+    this.partitionToAzureStore = new ConcurrentHashMap<>();
+    this.metricRegistry = metricRegistry;
+    this.azureMetrics = new AzureMetrics(metricRegistry);
+    this.cloudConfig = new CloudConfig(properties);
+    this.azureCloudConfig = new AzureCloudConfig(properties);
+    this.properties = properties;
+    this.clusterMap = clusterMap;
+    this.azureStorageClient =
+        new ConnectionStringBasedStorageClient(cloudConfig, azureCloudConfig, azureMetrics).getStorageSyncClient();
```

Comment:
Are you planning to make it configurable in later PRs since we are not using connection string based authentication in production because it is unsafe ([Reference PR](https://github.com/linkedin/ambry/pull/1679)). We are using either ADAuth based or ClientSecretCredential based.

Reply:
Yeah, if that is truly the case, I will make it configurable or just directly use the client we plan on using. But currently we are using connection-based client in vcr-video.